src/coreclr/jit/codegenlinear.cpp

```diff
@@ -366,6 +366,8 @@ void CodeGen::genCodeForBBlist()
 
             block->bbEmitCookie = GetEmitter()->emitAddLabel(gcInfo.gcVarPtrSetCur, gcInfo.gcRegGCrefSetCur,
                                                              gcInfo.gcRegByrefSetCur, FALSE);
+            JITDUMP("Mapped " FMT_BB " to G_M%03u_IG%02u\n", block->bbNum, compiler->compMethodID,
```

Comment:
Do we need this dump? We already dumped the block we're generating, and emitAddLabel dumps the IG#.

In particular, I don't want anyone outside of the emitter to access `emitCurIG`

Reply:
Yes, the mapping is spread out in the dump. 

```
=============== Generating BB01 [000..013), preds={} succs={BB02} flags=0x00000002.204b0020: i label target hascall gcsafe newobj LIR 
BB01 IN (1)={V99        } + ByrefExposed + GcHeap
     OUT(3)={V99 V03 V02} + ByrefExposed + GcHeap

Recording Var Locations at start of BB01
  <none>

Change life 000000000000000000000000000000000000000000000000 {} -> 000000000000000000000000000000000000800000000000 {V99}
							Live regs: (unchanged) 00000000 {}
							GC regs: (unchanged) 00000000 {}
							Byref regs: (unchanged) 00000000 {}

      L_M46582_BB01:
Label: IG02,
```

During diagnosis, it is easier to get all the mappings so we can relate the IGs to BBs. With my change, we can just search for "Mapped BB" in entire jit dump and get all the mappings. If we do not want to expose `emitCurIG`, I can probably pass `bbNum` as `DEBUG_ARG()` to `emitAddLabel()`?

Reply:
debug arg would be ok. Maybe you can change `BOOL isFinallyTarget` to `bool isFinallyTarget` at the same time.

But hopefully we don't have to pass `isFinallyTarget` at all call sites. Maybe add overloads:

```
 void* emitter::emitAddLabel(VARSET_VALARG_TP GCvars, regMaskTP gcrefRegs, bool isFinallyTarget = false, regMaskTP byrefRegs DEBUG_ARG(unsigned bbNum = 0))
{
...
}
 void* emitter::emitAddLabel(VARSET_VALARG_TP GCvars, regMaskTP gcrefRegs, regMaskTP byrefRegs DEBUG_ARG(unsigned bbNum))
{
  return emitAddLabel(GCvars, gcrefRegs, byrefRegs, false DEBUG_ARG(bbNum));
}
```

?

Reply:
Looks like `isFinallyTarget` is never `true` for Arm and thus will never set `IGF_FINALLY_TARGET`. Is that an oversight or did we stop using it at one point but didn't delete the code? @AndyAyersMS

Reply:
Alright, I deleted `IGF_FINALLY_TARGET` and code around it.

Reply:
I'm not familiar with `IGF_FINALLY_TARGET` -- Bruce probably knows.

Reply:
No, please don't delete the `IGF_FINALLY_TARGET` code.

Unfortunately, you have accidentally found a small GC hole in our arm32 codegen that has existed since the beginning of RyuJIT/arm32 (I presume the reason we added that code still exists). We need to call emitAddLabel in genCodeForBBlist() as:
```
block->bbEmitCookie = GetEmitter()->emitAddLabel(gcInfo.gcVarPtrSetCur, gcInfo.gcRegGCrefSetCur,
                                                 gcInfo.gcRegByrefSetCur,
#if FEATURE_EH_FUNCLETS && defined(TARGET_ARM)
                                           /*isFinally*/ block->bbFlags & BBF_FINALLY_TARGET
#else
                                           FALSE
#endif
                                            );
```

This code never got ported over from the legacy codegen. I wonder what the GC info diffs for arm32 will show...

Of course, this is a separate issue; you shouldn't do it as part of this change.

Reply:
I created https://github.com/dotnet/runtime/issues/51289 to track this.

Reply:
Thanks! I went ahead and reverted that commit. For now, I have just added `DEBUG_ARG(bbNum)` without creating an overloaded method. Hopefully when we address #51289 , we will do it properly.